synapse/lib/ast.py

```diff
@@ -1035,8 +1047,8 @@ async def run(self, runt, genr):
             item = [i async for i in s_stormtypes.toiter(item)]
 
             if len(item) < len(names):
-                mesg = 'Attempting to assign more items then we have variable to assign to.'
-                raise s_exc.StormVarListError(mesg=mesg, names=names, vals=item)
+                mesg = f'Attempting to assign more items than we have variables to assign to: {repr(item)[:256]}'
```

Comment:
Should catch that TypeError on these ones too

Reply:
Updated toiter to catch TypeErrors for non-prim values